src/main/java/seedu/address/model/entry/EntryContainsSearchTermsPredicate.java

```diff
@@ -44,9 +44,12 @@ public boolean test(Entry entry) {
         if (!result && findEntryDescriptor.getAll().isPresent()) {
             result = result
                 || entry.getTags().contains(new Tag(findEntryDescriptor.getAll().get()))
-                || StringUtil.containsPhraseIgnoreCase(entry.getTitle().fullTitle, findEntryDescriptor.getAll().get())
-                || StringUtil.containsPhraseIgnoreCase(entry.getDescription().value, findEntryDescriptor.getAll().get())
-                || StringUtil.containsPhraseIgnoreCase(entry.getLink().value, findEntryDescriptor.getAll().get());
+                || StringUtil.containsPhraseIgnoreCase(
+                        entry.getTitle().fullTitle, findEntryDescriptor.getAll().get())
+                || StringUtil.containsPhraseIgnoreCase(
+                        entry.getDescription().value, findEntryDescriptor.getAll().get())
+                || StringUtil.containsPhraseIgnoreCase(
+                        entry.getLink().value.toString(), findEntryDescriptor.getAll().get());
```

Comment:
Note that this kind of transformation happens a lot: we convert the links back to string format when doing any kind of textual manipulation, or when interfacing with WebEngine.